cmd/register/register_util.go

```diff
@@ -414,27 +441,97 @@ func getJSONSpec(message proto.Message) string {
+
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, "", err
+		}
+
+		err = json.Unmarshal(data, &FlyteSnacksRelease)
+		if err != nil {
+			return nil, "", err
+		}
+		return FlyteSnacksRelease, *releases[0].TagName, nil
+	}
+	return nil, "", fmt.Errorf("Repository doesn't have any release")
+}
+
+func getAdditionalDistributionLoc(remoteLocation, identifier string) storage.DataReference {
+	return storage.DataReference(fmt.Sprintf("%v/%v.tar.gz", remoteLocation, identifier))
+}
+
+func uploadFastRegisterArtifact(ctx context.Context, file, additionalDistributionDir, version string) error {
+	s, err := getStorageClient(ctx)
 	if err != nil {
-		return nil, "", err
+		return err
 	}
-	defer response.Body.Close()
 
-	data, err := ioutil.ReadAll(response.Body)
+	var dataRefReaderCloser io.ReadCloser
+	fullRemotePath := getAdditionalDistributionLoc(additionalDistributionDir, version)
+	raw, err := json.Marshal(file)
 	if err != nil {
-		return nil, "", err
+		return err
+	}
+	dataRefReaderCloser, err = os.Open(file)
+	if err != nil {
+		return err
+	}
+	dataRefReaderCloser, err = gzip.NewReader(dataRefReaderCloser)
+	if err != nil {
+		return err
+	}
+	if err := s.WriteRaw(ctx, fullRemotePath, int64(len(raw)), storage.Options{}, dataRefReaderCloser); err != nil {
+		return err
 	}
+	return nil
+}
 
-	err = json.Unmarshal(data, &FlyteSnacksRelease)
+func getStorageClient(ctx context.Context) (storage.ComposedProtobufStore, error) {
+	if Client != nil {
+		return Client, nil
+	}
+	testScope := promutils.NewTestScope()
+	// Set Keys
+	labeled.SetMetricKeys(contextutils.AppNameKey, contextutils.ProjectKey, contextutils.DomainKey)
+	s, err := storage.NewDataStore(storage.GetConfig(), testScope.NewSubScope("flytectl"))
 	if err != nil {
-		return nil, "", err
+		logger.Errorf(ctx, "error while creating storage client %v", err)
+		return nil, err
+	}
+	return s.ComposedProtobufStore, nil
+}
+
+func validateRegisterFiles(dataRefs []string) (string, []string, []string, error) {
+	var validProto, InvalidFiles []string
+	var sourceCode string
+	var isFast = false
+	for _, v := range dataRefs {
+		if len(rconfig.DefaultFilesConfig.AdditionalDistributionPath) > 0 && strings.HasSuffix(v, sourceCodeExtension) {
+			isFast = true
+			sourceCode = v
+		} else if len(rconfig.DefaultFilesConfig.AdditionalDistributionPath) == 0 && strings.HasSuffix(v, sourceCodeExtension) {
+			return sourceCode, validProto, InvalidFiles, fmt.Errorf("please pass additional flags like --additionalDistributionPath")
+		} else if strings.HasSuffix(v, ".pb") {
+			validProto = append(validProto, v)
+		} else {
+			InvalidFiles = append(InvalidFiles, v)
+		}
+	}
+	if !isFast && len(rconfig.DefaultFilesConfig.AdditionalDistributionPath) > 0 {
+		return sourceCode, validProto, InvalidFiles, fmt.Errorf("please check your fast serialize package, There is no source code available but you passed additional flag --additionalDistributionPath")
 	}
```

Comment:
this should just  be a warning(printed to o/p) and additional flags can be ignored.